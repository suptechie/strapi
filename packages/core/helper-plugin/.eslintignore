node_modules/
.eslintrc.js
build/
lib/src/utils/**
