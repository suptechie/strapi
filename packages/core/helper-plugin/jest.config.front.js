module.exports = {
  preset: '../../../jest-preset.front.js',
  collectCoverageFrom: ['<rootDir>/packages/core/helper-plugin/src/**/*.js'],
  displayName: 'Helper plugin',
};
