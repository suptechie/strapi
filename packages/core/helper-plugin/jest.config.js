const baseConfig = require('../../../jest.base-config')

module.exports = {
  displayName: 'helper-plugin',
  ...baseConfig,
  rootDir: '../../..',
  testMatch: [`<rootDir>/packages/core/helper-plugin${baseConfig.testMatch}`]
}
