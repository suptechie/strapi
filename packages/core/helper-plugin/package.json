{
  "name": "@strapi/helper-plugin",
  "version": "3.6.5",
  "description": "Helper for Strapi plugins development",
  "main": "build/index.js",
  "module": "build/index.esm.js",
  "files": [
    "build"
  ],
  "engines": {
    "node": ">=12.x.x <=16.x.x",
    "npm": ">=6.0.0"
  },
  "author": {
    "email": "hi@strapi.io",
    "name": "Strapi team",
    "url": "https://strapi.io"
  },
  "maintainers": [
    {
      "name": "Strapi team",
      "email": "hi@strapi.io",
      "url": "https://strapi.io"
    }
  ],
  "license": "SEE LICENSE IN LICENSE",
  "scripts": {
    "prepublishOnly": "npm run build",
    "prebuild": "rimraf build",
    "build": "rollup -c",
    "watch": "rollup -c -w",
    "test": "echo \"no tests yet\""
  },
  "peerDependencies": {
    "immer": "^8.0.1",
    "qs": "6.10.1",
    "react-select": "^4.0.2"
  },
  "devDependencies": {
    "@babel/core": "7.14.0",
    "@babel/plugin-proposal-class-properties": "7.13.0",
    "@babel/plugin-transform-runtime": "7.13.15",
    "@babel/preset-env": "7.14.0",
    "@babel/preset-react": "7.13.13",
    "@babel/runtime": "7.14.0",
    "@rollup/plugin-babel": "5.3.0",
    "@rollup/plugin-commonjs": "18.0.0",
    "@rollup/plugin-node-resolve": "11.2.1",
    "@rollup/plugin-replace": "2.4.2",
    "enzyme": "^3.8.0",
    "enzyme-adapter-react-16": "^1.15.6",
    "rimraf": "3.0.2",
    "rollup": "2.46.0",
    "rollup-plugin-node-polyfills": "0.2.1",
    "rollup-plugin-peer-deps-external": "2.2.4",
    "rollup-plugin-postcss": "4.0.0",
    "rollup-plugin-terser": "7.0.2"
  },
  "dependencies": {
    "@buffetjs/core": "3.3.5",
    "@buffetjs/custom": "3.3.5",
    "@buffetjs/hooks": "3.3.5",
    "@buffetjs/icons": "3.3.5",
    "@buffetjs/styles": "3.3.5",
    "@buffetjs/utils": "3.3.5",
    "@fortawesome/fontawesome-free": "^5.15.2",
    "@fortawesome/fontawesome-svg-core": "^1.2.35",
    "@fortawesome/free-brands-svg-icons": "^5.15.2",
    "@fortawesome/free-solid-svg-icons": "^5.15.3",
    "@fortawesome/react-fontawesome": "^0.1.14",
    "@rollup/plugin-image": "2.0.6",
    "axios": "0.21.1",
    "babel-plugin-styled-components": "1.12.0",
    "bootstrap": "^4.6.0",
    "classnames": "^2.3.1",
    "immutable": "^3.8.2",
    "invariant": "^2.2.1",
    "lodash": "4.17.21",
    "moment": "^2.29.1",
    "react": "^16.14.0",
    "react-dom": "^16.9.0",
    "react-helmet": "^6.1.0",
    "react-intl": "5.20.2",
    "react-router": "^5.2.0",
    "react-router-dom": "^5.0.0",
    "reactstrap": "8.4.1",
    "styled-components": "^5.2.3",
    "whatwg-fetch": "^3.6.2"
  },
  "gitHead": "231263a3535658bab1e9492c6aaaed8692d62a53"
}
