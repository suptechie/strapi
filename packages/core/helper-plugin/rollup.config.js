import peerDepsExternal from 'rollup-plugin-peer-deps-external';
import nodePolyfills from 'rollup-plugin-node-polyfills';
import babel from '@rollup/plugin-babel';
import { nodeResolve } from '@rollup/plugin-node-resolve';
import commonjs from '@rollup/plugin-commonjs';
import replace from '@rollup/plugin-replace';
import svg from 'rollup-plugin-svg';
import image from '@rollup/plugin-image';
import postcss from 'rollup-plugin-postcss';
import packageJson from './package.json';

export default {
  input: './lib/src/index.js',
  output: [
    {
      file: packageJson.main,
      format: 'cjs',
      sourcemap: false,
    },
    {
      file: packageJson.module,
      format: 'esm',
      sourcemap: false,
    },
  ],
  plugins: [
    peerDepsExternal({
      packageJsonPath: './package.json',
    }),
    image(),
    postcss({
      extensions: ['.css'],
    }),
    nodeResolve({
      extensions: ['.js'],
      preferBuiltins: true,
    }),
    replace({
      'process.env.NODE_ENV': JSON.stringify('production'),
      preventAssignment: true,
    }),
    babel({
      presets: [
        [
          '@babel/preset-env',
          {
            modules: false,
            targets: {
              browsers: ['Since 2017'],
            },
          },
        ],
        '@babel/preset-react',
      ],
      babelHelpers: 'runtime',
      plugins: [
        '@babel/plugin-transform-runtime',
        '@babel/plugin-proposal-class-properties',
        ['babel-plugin-styled-components', { pure: true }],
      ],
      exclude: 'node_modules/**',
    }),
    commonjs(),
    svg(),

    nodePolyfills(),
  ],
  external: [
    ...Object.keys(packageJson.dependencies || {}),
    ...Object.keys(packageJson.peerDependencies || {}),
  ],
};
