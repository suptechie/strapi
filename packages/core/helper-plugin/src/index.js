import { getType, getOtherInfos } from './content-manager/utils/getAttributeInfos';

/* -------------------------------------------------------------------------------------------------
 * Components
 * -----------------------------------------------------------------------------------------------*/

export { default as CheckPagePermissions } from './components/CheckPagePermissions';
export { default as CheckPermissions } from './components/CheckPermissions';
export { default as ConfirmDialog } from './components/ConfirmDialog';
export { default as ContentBox } from './components/ContentBox';
export { default as DynamicTable } from './components/DynamicTable';
export { default as EmptyStateLayout } from './components/EmptyStateLayout';
export { default as NoContent } from './components/NoContent';
export { default as NoMedia } from './components/NoMedia';
export { default as NoPermissions } from './components/NoPermissions';
export { default as AnErrorOccurred } from './components/AnErrorOccurred';
export { default as EmptyBodyTable } from './components/EmptyBodyTable';
export { default as GenericInput } from './components/GenericInput';
export * from './components/InjectionZone';
export { default as LoadingIndicatorPage } from './components/LoadingIndicatorPage';
export { default as NotAllowedInput } from './components/NotAllowedInput';
export { default as SettingsPageTitle } from './components/SettingsPageTitle';
export { default as SearchURLQuery } from './components/SearchURLQuery';
export { default as Status } from './components/Status';
export { default as FilterListURLQuery } from './components/FilterListURLQuery';
export { default as FilterPopoverURLQuery } from './components/FilterPopoverURLQuery';
export { default as Form } from './components/Form';
export { default as PaginationURLQuery } from './components/PaginationURLQuery';
export { default as PageSizeURLQuery } from './components/PageSizeURLQuery';
export { default as RelativeTime } from './components/RelativeTime';
export { default as DateTimePicker } from './components/DateTimePicker';
export { default as ReactSelect } from './components/ReactSelect';
export { default as Link } from './components/Link';
export { default as LinkButton } from './components/LinkButton';

/* -------------------------------------------------------------------------------------------------
 * Content Manager
 * -----------------------------------------------------------------------------------------------*/

export { default as ContentManagerEditViewDataManagerContext } from './content-manager/contexts/ContentManagerEditViewDataManagerContext';
export { default as useCMEditViewDataManager } from './content-manager/hooks/useCMEditViewDataManager';
export { getType };
export { getOtherInfos };

/* -------------------------------------------------------------------------------------------------
 * Features
 * -----------------------------------------------------------------------------------------------*/

export * from './features/AppInfo';
export * from './features/AutoReloadOverlayBlocker';
export * from './features/CustomFields';
export * from './features/GuidedTour';
export * from './features/Library';
export * from './features/OverlayBlocker';
export * from './features/Notifications';
export * from './features/RBAC';
export * from './features/StrapiApp';
export * from './features/Tracking';

/* -------------------------------------------------------------------------------------------------
 * Hooks
 * -----------------------------------------------------------------------------------------------*/

export { default as useQuery } from './hooks/useQuery';
export { useSelectionState } from './hooks/useSelectionState';
export * from './hooks/useAPIErrorHandler';
export { useFilter } from './hooks/useFilter';
export { useCollator } from './hooks/useCollator';
export { useCallbackRef } from './hooks/useCallbackRef';
export { useClipboard } from './hooks/useClipboard';

export { default as useQueryParams } from './hooks/useQueryParams';
export { default as useRBAC } from './hooks/useRBAC';
export { default as usePersistentState } from './hooks/usePersistentState';
export { default as useFocusWhenNavigate } from './hooks/useFocusWhenNavigate';
export { default as useLockScroll } from './hooks/useLockScroll';
export { default as useFetchClient } from './hooks/useFetchClient';

/* -------------------------------------------------------------------------------------------------
 * Icons
 * -----------------------------------------------------------------------------------------------*/

export { default as SortIcon } from './icons/SortIcon';
export { default as RemoveRoundedButton } from './icons/RemoveRoundedButton';

/* -------------------------------------------------------------------------------------------------
 * Utils
 * -----------------------------------------------------------------------------------------------*/

export { default as auth } from './utils/auth';
export { default as hasPermissions } from './utils/hasPermissions';
export { default as prefixFileUrlWithBackendUrl } from './utils/prefixFileUrlWithBackendUrl/prefixFileUrlWithBackendUrl';
export { default as prefixPluginTranslations } from './utils/prefixPluginTranslations';
export { default as pxToRem } from './utils/pxToRem';
export { default as to } from './utils/await-to-js';
export { default as setHexOpacity } from './utils/setHexOpacity';
export { default as translatedErrors } from './utils/translatedErrors';
export { default as formatContentTypeData } from './content-manager/utils/formatContentTypeData';
export { findMatchingPermissions } from './utils/hasPermissions';
export { default as contentManagementUtilRemoveFieldsFromData } from './content-manager/utils/contentManagementUtilRemoveFieldsFromData';
export { default as getFileExtension } from './utils/getFileExtension/getFileExtension';
export * from './utils/stopPropagation';
export { default as difference } from './utils/difference';
export { default as wrapAxiosInstance } from './utils/wrapAxiosInstance';

export { default as request } from './utils/request';
export { default as getAPIInnerErrors } from './utils/getAPIInnerErrors';
export { default as getYupInnerErrors } from './utils/getYupInnerErrors';

export { default as getFetchClient } from './utils/getFetchClient';
