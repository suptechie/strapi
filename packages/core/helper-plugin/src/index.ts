/* -------------------------------------------------------------------------------------------------
 * Components
 * -----------------------------------------------------------------------------------------------*/

export * from './components/CheckPermissions';
export * from './components/ConfirmDialog';
export * from './components/DynamicTable';
export * from './components/Table';
export * from './components/EmptyBodyTable';
export * from './components/FilterListURLQuery';
export * from './components/FilterPopoverURLQuery';
export * from './components/PageSizeURLQuery';
export * from './components/PaginationURLQuery';
export * from './components/RelativeTime';
export * from './components/SearchURLQuery';

/* -------------------------------------------------------------------------------------------------
 * Features
 * -----------------------------------------------------------------------------------------------*/

export * from './features/AppInfo';
export * from './features/AutoReloadOverlayBlocker';
export * from './features/CustomFields';
export * from './features/GuidedTour';
export * from './features/Library';
export * from './features/Notifications';
export * from './features/OverlayBlocker';
export * from './features/RBAC';
export * from './features/StrapiApp';
export * from './features/Tracking';

/* -------------------------------------------------------------------------------------------------
 * Hooks
 * -----------------------------------------------------------------------------------------------*/

export * from './hooks/useAPIErrorHandler';
export * from './hooks/useCallbackRef';
export * from './hooks/useClipboard';
export * from './hooks/useCollator';
export * from './hooks/useFetchClient';
export * from './hooks/useFilter';
export * from './hooks/useFocusInputField';
export * from './hooks/useFocusWhenNavigate';
export * from './hooks/useLockScroll';
export * from './hooks/usePersistentState';
export * from './hooks/useQuery';
export * from './hooks/useQueryParams';
export * from './hooks/useRBAC';
export * from './hooks/useSelectionState';

/* -------------------------------------------------------------------------------------------------
 * Utils
 * -----------------------------------------------------------------------------------------------*/

export * from './utils/auth';
export * from './utils/awaitToJs';
export * from './utils/difference';
export * from './utils/getAPIInnerErrors';
export * from './utils/getFetchClient';
export * from './utils/getFileExtension';
export * from './utils/getYupInnerErrors';
export * from './utils/hasPermissions';
export * from './utils/normalizeAPIError';
export * from './utils/prefixFileUrlWithBackendUrl';
export * from './utils/prefixPluginTranslations';
export * from './utils/pxToRem';
export * from './utils/request';
export * from './utils/setHexOpacity';
export * from './utils/stopPropagation';
export * from './utils/translatedErrors';
export * from './utils/wrapAxiosInstance';

/* -------------------------------------------------------------------------------------------------
 * Types
 * -----------------------------------------------------------------------------------------------*/

export type { TranslationMessage, FilterData } from './types';
