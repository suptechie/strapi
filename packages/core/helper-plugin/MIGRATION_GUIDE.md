# Helper-plugin Migration Guide

This document has been written incrementally to help developers migrate their strapi plugins & applications to _not_ use the `helper-plugin` package.
It is written in alphabetical order grouped by domain based on _every_ export that existed in the aforementioned package.

## Components

### DateTimePicker

This was aliasing the design-system. You should import the component from there:

```ts
// Before
import { DateTimePicker } from '@strapi/helper-plugin';

// After
import { DateTimePicker } from '@strapi/design-system';
```

### Link

This was aliasing the design-system and using the `as` prop with `react-router-dom`. You should import the component from there:

```ts
// Before
import { Link } from '@strapi/helper-plugin';

// After
import { Link } from '@strapi/design-system/v2';
import { NavLink } from 'react-router-dom';

const MyLink = () => {
  return (
    <Link as={NavLink} to="/my-link">
      My Link
    </Link>
  );
};
```

### LinkButton

This was aliasing the design-system and using the `as` prop with `react-router-dom`. You should import the component from there:

```ts
// Before
import { LinkButton } from '@strapi/helper-plugin';

// After
import { LinkButton } from '@strapi/design-system/v2';
import { NavLink } from 'react-router-dom';

const MyLink = () => {
  return (
    <LinkButton as={NavLink} to="/my-link">
      My Link
    </LinkButton>
  );
};
```

## Content Manager

### contentManagementUtilRemoveFieldsFromData

This function has been removed and not replaced. If you feel like you need this function, please open an issue on the Strapi repository to discuss your usecase.

### formatContentTypeData

This function has been removed and not replaced. If you feel like you need this function, please open an issue on the Strapi repository to discuss your usecase.

### useCMEditViewDataManager

This hook has been split into different hooks, each with more ability then it's previous:

- useDocument
- useDocumentLayout
- useDocumentRBAC
- useForm

```ts
// Before
import { useCMEditViewDataManager } from '@strapi/helper-plugin';

// After
import {
  useDocument,
  useDocumentActions,
  useDocumentLayout,
  useDocumentRBAC,
  useForm,
} from '@strapi/strapi/admin/hooks';
```

Some common use cases are listed below:

```ts
// Before
const { slug, isSingleType, isCreatingEntry } = useCMEditViewDataManager();

// After
const { model, id, collectionType } = useDocument();
const isSingleType = collectionType === 'single-types';
const isCreatingEntry = id === 'create';
```

```ts
// Before
const { onPublish, onUnpublish } = useCMEditViewDataManager();

// After
const { publish, unpublish } = useDocumentActions();
```

```ts
// Before
const { layout } = useCMEditViewDataManager();

// After
const {
  edit: { layout, components },
} = useDocumentLayout();
```
