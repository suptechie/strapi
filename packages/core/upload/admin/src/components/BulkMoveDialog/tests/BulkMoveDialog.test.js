import React from 'react';
import { ThemeProvider, lightTheme } from '@strapi/design-system';
import { IntlProvider } from 'react-intl';
import { render } from '@testing-library/react';
import { NotificationsProvider } from '@strapi/helper-plugin';
import { QueryClientProvider, QueryClient } from 'react-query';

import { BulkMoveDialog } from '..';

jest.mock('../../../hooks/useFolderStructure');

const client = new QueryClient({
  defaultOptions: {
    queries: {
      retry: false,
    },
  },
});

function ComponentFixture(props) {
  return (
    <QueryClientProvider client={client}>
      <IntlProvider locale="en" messages={{}}>
        <ThemeProvider theme={lightTheme}>
          <NotificationsProvider toggleNotification={() => {}}>
            <BulkMoveDialog {...props} />
          </NotificationsProvider>
        </ThemeProvider>
      </IntlProvider>
    </QueryClientProvider>
  );
}

function setup(props = { onClose: jest.fn() }) {
  return render(<ComponentFixture {...props} />, { container: document.body });
}

describe('BulkMoveDialog', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  test('renders and matches the snapshot', () => {
    const { container } = setup({
      selected: [],
      onClose: jest.fn(),
    });
    expect(container).toMatchSnapshot();
  });
});
