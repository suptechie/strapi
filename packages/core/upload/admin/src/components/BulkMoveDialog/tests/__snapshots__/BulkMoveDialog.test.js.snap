// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BulkMoveDialog renders and matches the snapshot 1`] = `
.c0 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c15 {
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c16 > * {
  margin-top: 0;
  margin-bottom: 0;
}

.c16 > * + * {
  margin-top: 4px;
}

.c7 {
  font-weight: 600;
  color: #32324d;
  font-size: 0.875rem;
  line-height: 1.43;
}

.c28 {
  font-weight: 600;
  color: #32324d;
  font-size: 0.75rem;
  line-height: 1.33;
}

.c25 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  cursor: pointer;
  padding: 8px;
  border-radius: 4px;
  background: #ffffff;
  border: 1px solid #dcdce4;
  position: relative;
  outline: none;
}

.c25 svg {
  height: 12px;
  width: 12px;
}

.c25 svg > g,
.c25 svg path {
  fill: #ffffff;
}

.c25[aria-disabled='true'] {
  pointer-events: none;
}

.c25:after {
  -webkit-transition-property: all;
  transition-property: all;
  -webkit-transition-duration: 0.2s;
  transition-duration: 0.2s;
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -4px;
  bottom: -4px;
  left: -4px;
  right: -4px;
  border: 2px solid transparent;
}

.c25:focus-visible {
  outline: none;
}

.c25:focus-visible:after {
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -5px;
  bottom: -5px;
  left: -5px;
  right: -5px;
  border: 2px solid #4945ff;
}

.c26 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 8px 16px;
  background: #4945ff;
  border: 1px solid #4945ff;
  border: 1px solid #dcdce4;
  background: #ffffff;
}

.c26 .sc-cfJLRR {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c26 .c27 {
  color: #ffffff;
}

.c26[aria-disabled='true'] {
  border: 1px solid #dcdce4;
  background: #eaeaef;
}

.c26[aria-disabled='true'] .c27 {
  color: #666687;
}

.c26[aria-disabled='true'] svg > g,
.c26[aria-disabled='true'] svg path {
  fill: #666687;
}

.c26[aria-disabled='true']:active {
  border: 1px solid #dcdce4;
  background: #eaeaef;
}

.c26[aria-disabled='true']:active .c27 {
  color: #666687;
}

.c26[aria-disabled='true']:active svg > g,
.c26[aria-disabled='true']:active svg path {
  fill: #666687;
}

.c26:hover {
  background-color: #f6f6f9;
}

.c26:active {
  background-color: #eaeaef;
}

.c26 .c27 {
  color: #32324d;
}

.c26 svg > g,
.c26 svg path {
  fill: #32324d;
}

.c29 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 8px 16px;
  background: #4945ff;
  border: 1px solid #4945ff;
}

.c29 .sc-cfJLRR {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c29 .c27 {
  color: #ffffff;
}

.c29[aria-disabled='true'] {
  border: 1px solid #dcdce4;
  background: #eaeaef;
}

.c29[aria-disabled='true'] .c27 {
  color: #666687;
}

.c29[aria-disabled='true'] svg > g,
.c29[aria-disabled='true'] svg path {
  fill: #666687;
}

.c29[aria-disabled='true']:active {
  border: 1px solid #dcdce4;
  background: #eaeaef;
}

.c29[aria-disabled='true']:active .c27 {
  color: #666687;
}

.c29[aria-disabled='true']:active svg > g,
.c29[aria-disabled='true']:active svg path {
  fill: #666687;
}

.c29:hover {
  border: 1px solid #7b79ff;
  background: #7b79ff;
}

.c29:active {
  border: 1px solid #4945ff;
  background: #4945ff;
}

.c29 svg > g,
.c29 svg path {
  fill: #ffffff;
}

.c19 {
  padding-right: 12px;
}

.c18 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c17 {
  font-weight: 600;
  color: #32324d;
  font-size: 0.75rem;
  line-height: 1.33;
}

.c20 {
  background: transparent;
  border: none;
  position: relative;
  z-index: 1;
}

.c20 svg {
  height: 0.6875rem;
  width: 0.6875rem;
}

.c20 svg path {
  fill: #666687;
}

.c21 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background: none;
  border: none;
}

.c21 svg {
  width: 0.5625rem;
}

.c13 {
  display: grid;
  grid-template-columns: repeat(12,1fr);
  gap: 16px;
}

.c14 {
  grid-column: span 12;
  max-width: 100%;
}

.c2 {
  background: #ffffff;
  border-radius: 4px;
  box-shadow: 0px 2px 15px rgba(33,33,52,0.1);
}

.c4 {
  background: #f6f6f9;
  padding-top: 16px;
  padding-right: 20px;
  padding-bottom: 16px;
  padding-left: 20px;
}

.c11 {
  padding: 32px;
}

.c1 {
  position: fixed;
  z-index: 4;
  inset: 0;
  background: #32324d1F;
  padding: 0 40px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c3 {
  width: 51.875rem;
}

.c6 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c23 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c10 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  cursor: pointer;
  padding: 8px;
  border-radius: 4px;
  background: #ffffff;
  border: 1px solid #dcdce4;
  position: relative;
  outline: none;
}

.c8 svg {
  height: 12px;
  width: 12px;
}

.c8 svg > g,
.c8 svg path {
  fill: #ffffff;
}

.c8[aria-disabled='true'] {
  pointer-events: none;
}

.c8:after {
  -webkit-transition-property: all;
  transition-property: all;
  -webkit-transition-duration: 0.2s;
  transition-duration: 0.2s;
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -4px;
  bottom: -4px;
  left: -4px;
  right: -4px;
  border: 2px solid transparent;
}

.c8:focus-visible {
  outline: none;
}

.c8:focus-visible:after {
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -5px;
  bottom: -5px;
  left: -5px;
  right: -5px;
  border: 2px solid #4945ff;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 2rem;
  width: 2rem;
}

.c9 svg > g,
.c9 svg path {
  fill: #8e8ea9;
}

.c9:hover svg > g,
.c9:hover svg path {
  fill: #666687;
}

.c9:active svg > g,
.c9:active svg path {
  fill: #a5a5ba;
}

.c9[aria-disabled='true'] {
  background-color: #eaeaef;
}

.c9[aria-disabled='true'] svg path {
  fill: #666687;
}

.c5 {
  border-radius: 4px 4px 0 0;
  border-bottom: 1px solid #eaeaef;
}

.c22 {
  border-radius: 0 0 4px 4px;
  border-top: 1px solid #eaeaef;
}

.c24 > * + * {
  margin-left: 8px;
}

.c12 {
  overflow: auto;
  max-height: 60vh;
}

@media (max-width:68.75rem) {
  .c14 {
    grid-column: span;
  }
}

@media (max-width:34.375rem) {
  .c14 {
    grid-column: span 12;
  }
}

<body
  class="lock-body-scroll"
>
  <div>
    <div
      class="c0"
    >
      <p
        aria-live="polite"
        aria-relevant="all"
        id="live-region-log"
        role="log"
      />
      <p
        aria-live="polite"
        aria-relevant="all"
        id="live-region-status"
        role="status"
      />
      <p
        aria-live="assertive"
        aria-relevant="all"
        id="live-region-alert"
        role="alert"
      />
    </div>
  </div>
  <div
    data-react-portal="true"
  >
    <div
      class="c1"
    >
      <div>
        <div>
          <div
            aria-labelledby="title"
            aria-modal="true"
            class="c2 c3"
            role="dialog"
          >
            <form
              action="#"
              novalidate=""
            >
              <div
                class="c4 c5"
              >
                <div
                  class="c6"
                >
                  <h2
                    class="c7"
                    id="title"
                  >
                    Move elements to
                  </h2>
                  <button
                    aria-disabled="false"
                    class="c8 c9"
                    type="button"
                  >
                    <span
                      class="c10"
                    >
                      Close the modal
                    </span>
                    <svg
                      aria-hidden="true"
                      fill="none"
                      focusable="false"
                      height="1em"
                      viewBox="0 0 24 24"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M24 2.417L21.583 0 12 9.583 2.417 0 0 2.417 9.583 12 0 21.583 2.417 24 12 14.417 21.583 24 24 21.583 14.417 12 24 2.417z"
                        fill="#212134"
                      />
                    </svg>
                  </button>
                </div>
              </div>
              <div
                class="c11 c12"
              >
                <div
                  class="c13"
                >
                  <div
                    class="c14"
                  >
                    <div
                      class=""
                    >
                      <div
                        class="c15 c16"
                        spacing="1"
                      >
                        <label
                          class="c17"
                          for="folder-destination"
                        >
                          <div
                            class="c18"
                          >
                            Location
                          </div>
                        </label>
                        <div
                          class=" css-1iyvlxg-container"
                        >
                          <span
                            class="css-1f43avz-a11yText-A11yText"
                            id="react-select-2-live-region"
                          />
                          <span
                            aria-atomic="false"
                            aria-live="polite"
                            aria-relevant="additions text"
                            class="css-1f43avz-a11yText-A11yText"
                          />
                          <div
                            class=" css-1e1stxh-control"
                          >
                            <div
                              class=" css-do2euy-ValueContainer"
                            >
                              <div
                                class=" css-1sqm0gs-singleValue"
                              >
                                Media Library
                              </div>
                              <input
                                aria-autocomplete="list"
                                aria-expanded="false"
                                aria-haspopup="true"
                                aria-invalid="false"
                                aria-readonly="true"
                                class="css-mohuvp-dummyInput-DummyInput"
                                id="folder-destination"
                                inputmode="none"
                                role="combobox"
                                tabindex="0"
                                value=""
                              />
                            </div>
                            <div
                              class=" css-1hb7zxy-IndicatorsContainer"
                            >
                              <button
                                class="c19 c20 c21"
                                type="button"
                              >
                                <svg
                                  fill="none"
                                  height="1em"
                                  viewBox="0 0 14 8"
                                  width="1em"
                                  xmlns="http://www.w3.org/2000/svg"
                                >
                                  <path
                                    clip-rule="evenodd"
                                    d="M14 .889a.86.86 0 01-.26.625L7.615 7.736A.834.834 0 017 8a.834.834 0 01-.615-.264L.26 1.514A.861.861 0 010 .889c0-.24.087-.45.26-.625A.834.834 0 01.875 0h12.25c.237 0 .442.088.615.264a.86.86 0 01.26.625z"
                                    fill="#32324D"
                                    fill-rule="evenodd"
                                  />
                                </svg>
                              </button>
                            </div>
                          </div>
                          <input
                            name="destination"
                            type="hidden"
                            value=""
                          />
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="c4 c22"
              >
                <div
                  class="c6"
                >
                  <div
                    class="c23 c24"
                  >
                    <button
                      aria-disabled="false"
                      class="c25 c26"
                      name="cancel"
                      type="button"
                    >
                      <span
                        class="c27 c28"
                      >
                        Cancel
                      </span>
                    </button>
                  </div>
                  <div
                    class="c23 c24"
                  >
                    <button
                      aria-disabled="false"
                      class="c25 c29"
                      type="submit"
                    >
                      <span
                        class="c27 c28"
                      >
                        Move
                      </span>
                    </button>
                  </div>
                </div>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
